### DISCUSSION TITLE
SunbirdED-Portal :  Not able to consume EPUB contents

### DISCUSSION THREAD

**Discussion by princegupta1131** [Sunbird-Knowlg/Community#133] [Bugs]
**SunbirdED-Portal :  Not able to consume EPUB contents facing below issue** 
1. Page count is not working.
2. Unbale to move to the next page
3.  Screen is blinking
4. Continues service is triggering.
Package Version Used => "@project-sunbird/sunbird-epub-player-v9": "5.5.0",
Portal link => https://staging.sunbirded.org/resources/play/content/do_21369950530254438411092

can anyone please look into this Issue 
CC: @HarishGangula @pallakartheekreddy @rajeevsathish @vinukumar-vs @Krishnaj20

Note: Please check the reference videos attached

https://github.com/Sunbird-Knowlg/Community/assets/114015020/83835b5e-e8c1-42e0-97b5-ea6e0ab14817

https://github.com/Sunbird-Knowlg/Community/assets/114015020/b2172a3a-5781-461d-8cab-35cfb769f419

![image](https://github.com/Sunbird-Knowlg/Community/assets/114015020/d41e88d4-2fe2-44d0-97f7-20e98cdbd413)

**Comment by HarishGangula** [comment]
@princegupta1131 can you try reproducing the issue here  https://github.com/Sunbird-Knowlg/sunbird-epub-player/tree/release-5.5.0,

steps to run are 
1. npm install
2. update the metadata in app.component.ts file with the above attached content
3. npm run start
4. open localhost:4000 in the browser and check

compare the and see if there are differences in input to player other dependencies

**Reply by princegupta1131** [reply]
Hi @HarishGangula, I have followed the above steps that you have mentioned.

1.  Standalone application content is loading. but count having some issues.
2. in sunbird portal used the same version but on click of the next page button after 2-3 pages it's not moving forward.
3. for some content screens are blinking and multiple blob calls in the network.

**4. The issue is occurring while clicking the next page button**  (playerEvent(),(telemetryEvent())

https://github.com/Sunbird-Knowlg/Community/assets/114015020/55f00d65-42bd-4cb7-a7d9-a3a2c964c2df

**Reply by HarishGangula** [reply]
@princegupta1131 Can you check and compare this standalone app and portal code that is there any difference for player to see the issue

**Comment by princegupta1131** [comment]
Hi @HarishGangula I tried with the same configuration of the demo app in the portal, but still next button click is not working properly. can you please suggest how we can fix this issue? 

CC: @rajeevsathish @vinukumar-vs

**Comment by vaivk369** [comment]
@HarishGangula In coKreat also we are facing the same issue. Please help.

**Reply by HarishGangula** [reply]
@vaivk369 @princegupta1131  Can you integrate the web component version of the player and see if you can reproduce it?
https://www.npmjs.com/package/@project-sunbird/sunbird-epub-player-web-component

**Comment by princegupta1131** [comment]
Hi, @HarishGangula post integration of the web component version of the player.  getting the same behaviour. The next button click is not working properly.

**Comment by vaivk369** [comment]
@HarishGangula After using as a web component, for me also, the pages are flickering and after traversing some pages, the next button does not load new page.

**Comment by HarishGangula** [comment]
@vaivk369 @princegupta1131  Can you check that this is specific to epub content or for all the epub contents are having this issue?
if it is specific epub content then share the analysis where this issue is triggered in the code

**Comment by vaivk369** [comment]
I am facing for all epub contents.

**Reply by HarishGangula** [reply]
@vaivk369  Did you tried https://github.com/orgs/Sunbird-Knowlg/discussions/133#discussioncomment-6118460 if not please check and udpate

**Comment by vaivk369** [comment]
As per the discussion happened, for now we need to fix the issue by applying css in the global css of portal where the player is integrated. 

Fixed in the coKreat with the code : [https://github.com/Sunbird-coKreat/creation-portal/pull/2221](https://github.com/Sunbird-coKreat/creation-portal/pull/2221)
@princegupta1131 as you are also facing the same issue, can you please check and then mark this thread as answered?

**Reply by princegupta1131** [reply]
Hi Team,
The Epub player issue fixed by adding the following code.

**PR link for reference** => https://github.com/Sunbird-Ed/SunbirdEd-portal/pull/8798/files

**Jira Ticket** => https://project-sunbird.atlassian.net/browse/ED-2199

CC: @rajeevsathish @vinukumar-vs

**Reply by vinukumar-vs** [reply]
@vaivk369 @princegupta1131 Is the fix required on all the adopters portal or is it required on Epub player? 

I dont see the details why did you changed this CSS on portal?

@HarishGangula Can you also confirm why this change is must & required on all the EPUB integration portals?

@pallakartheekreddy @rajeevsathish @Krishnaj20

**Reply by princegupta1131** [reply]
@vinukumar-vs This CSS code is added because of on click of the next page button. the screen was flickering and unable to proceed. as we compared it with the working epub in the Sunbird-knowledge platform. we found that additional CSS was written inside the 
**epub-viewer tag** compared to the portal. so we added the CSS and it worked.

CC: @HarishGangula  @pallakartheekreddy @rajeevsathish @Krishnaj20

**Reply by vinukumar-vs** [reply]
@vaivk369 @princegupta1131 I am still not clear with the details that you guys added here? 

Confirm is the change required for all adopters of EPUB player or it should be on epub-player itself? 

@HarishGangula @pallakartheekreddy please verify on your end as well, why all the adopters has to fix this CSS issue? Why can't it be done on EPUB player itself.

@Krishnaj20 @mmanimegalai [@ashokreddy1208](https://github.com/ashokreddy1208) [@chitranshu-keshav](https://github.com/chitranshu-keshav) [@rajeevsathish](https://github.com/rajeevsathish) [@swayangjit](https://github.com/swayangjit)

**Reply by HarishGangula** [reply]
@vinukumar-vs  We will publish the new player version with the fix, as of now this is a quick workaround for it.
cc: @pallakartheekreddy
